[DataGrid] Add default background color to grid (#16066)
mui/mui-x: docs/data/data-grid/style/style.md

{{"demo": "StripedGrid.js", "bg": "inline"}}

> ## Theme container, header and pinned sections

Reviewer: Oxford comma. 😁 and is "Theme" necessary here? If not I think it's better without, for the sake of keeping the header as short as possible. I'm on the fence about whether or not to pluralize everything so I'll leave that to your judgement.

```suggestion
## Containers, headers, and pinned sections
```

By default, the Data Grid uses the Material UI `theme.palette.background.default` color for the background color of the grid container, the column headers, and the pinned rows and columns.

> The various background colors can be overridden with the following configuration:

Reviewer: passive –> active voice

```suggestion
You can override these background colors with the following theme configuration:
```

```tsx
import { createTheme } from '@mui/material/styles';
import type {} from '@mui/x-data-grid/themeAugmentation';

const theme = createTheme({
  palette: {
    DataGrid: {
      // Container background
      bg: '#f8fafc',
      // Pinned rows and columns background
      pinnedBg: '#f1f5f9',
      // Column header background

>       headerBg: '#eaeff5',

Author: This is another breaking change to make Data Grid theming consistent with Material UI.

Instead of using the `mixins` theme property, we add `DataGrid` to the `palette`—in addition to being consistent with other [component color options](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/styles/createThemeWithVars.d.ts#L220-L235), this gives us built-in support for supporting light/dark mode theming with [`colorSchemes`](https://mui.com/material-ui/customization/palette/#color-schemes).

Material use `bg` instead of `background`, so also updated the property names for consistency.

    },
  },
});
```

> Material UI v6 users can use the `colorSchemes` property to specify different colors for light and dark mode:

Reviewer: might be nice to add h3s for `### Material UI v6` and `### Material UI v5` so users can quickly differentiate between the two samples when skimming

Author: Makes sense! The sections are quite specific to light and dark mode theming so I've added that to the headings:

- Light and dark mode in Material UI v5
- Light and dark mode in Material UI v5

const theme = createTheme({
  mixins: {
    MuiDataGrid: {
      // Container background

>       background: '#fafaf9',

Reviewer 2: How to handle dark mode here? It would be great to have this included in this demo.

Author: Mentioned here https://github.com/mui/mui-x/pull/16066#discussion_r1928401861
